api/src/main/java/marquez/db/NamespaceDao.java

```diff
@@ -78,7 +78,27 @@ default Namespace upsertNamespaceMeta(
   @SqlQuery("SELECT * FROM namespaces ORDER BY name LIMIT :limit OFFSET :offset")
   List<Namespace> findAll(int limit, int offset);
 
-  @SqlQuery(
+  default NamespaceRow upsertNamespaceRow(
+      UUID uuid, Instant now, String name, String currentOwnerName) {
+    doUpsertNamespaceRow(uuid, now, name, currentOwnerName);
+    return findNamespaceByName(name).orElseThrow();
```

Comment:
_Minor_: The `RETURNING *` was to avoid inserting, then querying for the namespace in two separate queries. But, also understand the cost of always updating `updated_at` to trigger the modified row to be returned. At some point, I'd love to flush out the usage of `updated_at` for all of our tables.